js/autotrack/__tests__/heapIgnore.spec.js

```diff
@@ -0,0 +1,206 @@
+        </Foo>
+      );
+      const normalComponent = wrapper
+        .find({ testID: 'targetElement' })
+        .filter(Text);
+      const normalProps = getBaseComponentProps(normalComponent.instance());
+      expect(normalProps).toEqual({
+        targetText: 'foobar',
+        touchableHierarchy:
+          'WrapperComponent;|Foo;|BarClass;|BarFunction;|Text;[testID=targetElement];|',
+      });
+    });
+
+    it('Ignores interaction', () => {
+      const wrapper = mount(
+        <Foo>
+          <HeapIgnore>
+            <Text testID="targetElement">{'foobar'}</Text>
+          </HeapIgnore>
+        </Foo>
+      );
+      const normalComponent = wrapper
+        .find({ testID: 'targetElement' })
+        .filter(Text);
+      const normalProps = getBaseComponentProps(normalComponent.instance());
+      expect(normalProps).toBe(null);
+    });
+
+    it('Ignores inner hierarchy', () => {
+      const wrapper = mount(
+        <Foo>
+          <HeapIgnore ignoreInteraction={false} ignoreTargetText={false}>
+            <Text testID="targetElement">{'foobar'}</Text>
+          </HeapIgnore>
+        </Foo>
+      );
+      const normalComponent = wrapper
+        .find({ testID: 'targetElement' })
+        .filter(Text);
+      const normalProps = getBaseComponentProps(normalComponent.instance());
+      expect(normalProps).toEqual({
+        targetText: 'foobar',
+        touchableHierarchy:
+          'WrapperComponent;|Foo;|BarClass;|BarFunction;|HeapIgnore;|',
+      });
+    });
+
+    it('Ignores props', () => {
+      const wrapper = mount(
+        <Foo>
+          <HeapIgnore
+            ignoreInteraction={false}
+            ignoreInnerHierarchy={false}
+            ignoreTargetText={false}
+          >
+            <Text testID="targetElement">{'foobar'}</Text>
+          </HeapIgnore>
+        </Foo>
+      );
+      const normalComponent = wrapper
+        .find({ testID: 'targetElement' })
+        .filter(Text);
+      const normalProps = getBaseComponentProps(normalComponent.instance());
+      expect(normalProps).toEqual({
+        targetText: 'foobar',
+        touchableHierarchy:
+          'WrapperComponent;|Foo;|BarClass;|BarFunction;|HeapIgnore;|Text;|',
+      });
+    });
+
+    it('Ignores target text', () => {
+      const wrapper = mount(
+        <Foo>
+          <HeapIgnore
+            ignoreInteraction={false}
+            ignoreInnerHierarchy={false}
+            ignoreAllProps={false}
+          >
+            <Text testID="targetElement">{'foobar'}</Text>
+          </HeapIgnore>
+        </Foo>
+      );
+      const normalComponent = wrapper
+        .find({ testID: 'targetElement' })
+        .filter(Text);
+      const normalProps = getBaseComponentProps(normalComponent.instance());
+      expect(normalProps).toEqual({
+        touchableHierarchy:
+          'WrapperComponent;|Foo;|BarClass;|BarFunction;|HeapIgnore;|Text;[testID=targetElement];|',
+      });
+    });
+
+    it('No-op', () => {
```

Comment:
Can you give a more descriptive name for the test?  Is there a condition under which it should be a no-op?  I can imagine this test suddenly breaking months later because of cosmic rays, and having no idea what thing we should be looking for.

Reply:
Ah yea, good catch.  Changed the `it` name and also added a TODO to the `HeapIgnore` component to add any other capture restrictions to it when we add them to `HeapIgnore` functionality (e.g. if we start capturing a new property, we'd probably add a prop to the HeapIgnore config, and we'd want to add it here to keep this a test for ignoring nothing).